internal/functions/efgs/upload-batch.go

```diff
@@ -189,7 +189,7 @@ func uploadBatch(ctx context.Context, batch *efgsapi.DiagnosisKeyBatch, config *
 		return &parsedResponse, nil
 	default:
 		logger.Debug("Batch upload failed. No key was uploaded")
-		return &parsedResponse, fmt.Errorf("HTTP %v: %v", res.StatusCode, string(body))
```

Comment:
I see the bug you're fixing here, however, I believe you just created another (possible) one. Please revert this change and just add a branch for HTTP 400 case.